### ISSUE TITLE
Better document type_subtype_pairs ege* parameter

### ISSUE THREAD

**Issue by hypeJunction** [Elgg/Elgg#8850]
The inline docs for `type_subtype_pairs` param in `elgg_get_entities()` are rather ambiguous. One may conclude that the array should have a form of:

```php
$options['type_subtype_pairs'] = array(
  array('object' => 'blog'),
  array('object' => 'file'),
);
```

as opposed to 

```php
$options['type_subtype_pairs'] = array(
   'object' => array('blog', 'file'),
);
```

**Comment by mrclay** [comment]
Seems like we could log a warning if the `$options['type_subtype_pairs']` keys are ints. But definitely improve docs.

**Comment by jdalsem** [comment]
Fixed in #8890
